tools/c7n_azure/tests_azure/test_azure_utils.py

```diff
@@ -280,6 +281,24 @@ def test_get_keyvault_secret(self, _1):
 
             result = get_keyvault_secret(None, 'https://testkv.vault.net/secrets/testsecret/123412')
             self.assertEqual(result, mock.value)
+            resource_auth = _1.call_args.kwargs.get('resource')
+            self.assertEqual(get_keyvault_auth_endpoint(AZURE_PUBLIC_CLOUD), resource_auth)
+
+    @patch('msrestazure.azure_active_directory.MSIAuthentication')
+    def test_get_keyvault_secret_with_parameter(self, _1):
+        mock = Mock()
+        mock.value = '{"client_id": "client", "client_secret": "secret"}'
+        with patch('azure.common.credentials.ServicePrincipalCredentials.__init__',
+                   return_value=None), \
+                patch('azure.keyvault.v7_0.KeyVaultClient.get_secret', return_value=mock):
+
+            reload(sys.modules['c7n_azure.utils'])
+
+            result = get_keyvault_secret(None, 'https://testkv.vault.net/secrets/testsecret/123412',
+                                         cloud_endpoints=AZURE_CHINA_CLOUD)
+            self.assertEqual(result, mock.value)
+            resource_auth = _1.call_args.kwargs.get('resource')
+            self.assertEqual(get_keyvault_auth_endpoint(AZURE_CHINA_CLOUD), resource_auth)
```

Comment:
I think it is better to construct an expected endpoint manually. Since `get_keyvault_secret` uses same function, if you accidentally edit the function test will still be successful.